__tests__/app.test.js

```diff
@@ -153,13 +152,32 @@ describe('4 get/api/articles/:article_id', () => {
             body : "Power of Water, Powers Unite! Six working together to fight evil!"
         }
         return request(app).post("/api/articles/stonks/comments").expect(400).send(comment).then((res)=>{
-            expect(res.body.msg).toBe("Bad, bad Request")
+            expect(res.body.msg).toBe("Thats not gonna work, dude. Try inserting a number")
         })
     })
     test('returns 400, if client tries to insert an empty object', () => {
         const comment = {}
-        return request(app).post("/api/articles/stonks/comments").expect(400).send(comment).then((res)=>{
+        return request(app).post("/api/articles/4/comments").expect(400).send(comment).then((res)=>{
             expect(res.body.msg).toBe("Bad, bad Request")
         })
     })
+    test('returns 404 if wanting to post in an article that doesnt exist', () => {
+        const comment = {
+            author: "lurker",
+            body : "Power of Water, Powers Unite! Six working together to fight evil!"
+        }
+        return request(app).post("/api/articles/9999999/comments").expect(404).send(comment).then((res)=>{
+            expect(res.body.msg).toBe("article not found!")
+        })
+    })
+    test('returns 400 if wanting to post from a user that doesnt exist', () => {
+        const comment = {
+            author: "Blue power Ranger",
+            body : "Power of Water, Powers Unite! Six working together to fight evil!"
+        }
+        return request(app).post("/api/articles/4/comments").expect(404).send(comment).then((res)=>{
+    
+            expect(res.body.msg).toBe("author is nonexistant")
+        })
+    });;
   });
```

Comment:
All of these are valid tests but what would happen if you gave a comment with a valid author but not a body? This should be a separate test.